src/core/net/tcp6.cpp

```diff
@@ -95,40 +125,69 @@ const SockAddr &Tcp::Endpoint::GetPeerAddress(void) const
 
 Error Tcp::Endpoint::Bind(const SockAddr &aSockName)
 {
-    OT_UNUSED_VARIABLE(aSockName);
+    Error error;
+    VerifyOrExit(!static_cast<const Address *>(&aSockName.mAddress)->IsUnspecified(), error = kErrorInvalidArgs);
+    VerifyOrExit(GetInstance().Get<Tcp>().CanBind(aSockName), error = kErrorInvalidState);
 
-    return kErrorNotImplemented;
+    memcpy(&mTcb.laddr, &aSockName.mAddress, sizeof(mTcb.laddr));
+    mTcb.lport = HostSwap16(aSockName.mPort);
+    error      = kErrorNone;
+
+exit:
+    return error;
 }
 
 Error Tcp::Endpoint::Connect(const SockAddr &aSockName, uint32_t aFlags)
 {
-    OT_UNUSED_VARIABLE(aSockName);
     OT_UNUSED_VARIABLE(aFlags);
 
-    return kErrorNotImplemented;
+    Error error = kErrorNone;
+
+    VerifyOrExit(mTcb.t_state == TCP6S_CLOSED, error = kErrorInvalidState);
+
+    struct sockaddr_in6 sin6p;
```

Comment:
Move variable declaration to the top of scope?

From our style guide:

> - Scoped Variable declarations
>   - All scoped (i.e. stack) variable declarations should be placed together at the top of the enclosing scope in which they are used.
>   - There shall be an empty line after all such variable declarations.
>   - The names of all variable declarations should be left aligned.

Reply:
OK. Fixed in the latest update.